### DISCUSSION TITLE
Swipe to dismiss

### DISCUSSION THREAD

**Discussion by wojtek717** [wojtek717/bottom-sheet#18] [General]
Should bottom sheet support "swipe to dismiss"?

1. When user swipes bottom sheet below some point we can close sheet
2. It would be enabled/disabled by some view modifier

@samdogg7 @kbabis 
What do you think guys? Should we support this behaviour?

**Comment by samdogg7** [comment]
I think this would be a nice addition in conjunction with the implementation of detent bindings. 

How would the swipe to dismiss behavior work? Would we have add a detent of `0`? Allow the drag gesture ended method to determine if it should go to zero?

**Reply by wojtek717** [reply]
If this swipe to dismiss would be enabled we can somehow modify "clampedDesiredHeight" minimum to reach this point

`    func dragGestureOnChanged(_ gesture: DragGesture.Value, screenHeight: CGFloat) {
        let desiredHeight = sheetHeight - gesture.translation.height
        
        print(desiredHeight)

        guard desiredHeight != sheetHeight else { return }

        let minHeight = minHeight(for: screenHeight)
        let maxHeight = maxFraction * screenHeight

        // Clamp desired height within bounds
        let clampedDesiredHeight = max(minHeight, min(desiredHeight, maxHeight))
        
        // Smooth the transition between the current height and the target
        sheetHeight = (sheetHeight * (1 - smoothingFactor)) + (clampedDesiredHeight * smoothingFactor)
    }`

**Reply by wojtek717** [reply]
Detention 0 or a little below is not a bad idea because the user would not have to reach exactly that value just get closer than .small detention.
